runner/src/rpc/api.rs

```diff
@@ -119,6 +129,54 @@ pub async fn view(
     Ok(serde_call_result)
 }
 
+pub async fn view_state(
+    contract_id: AccountId,
+    prefix: Option<StoreKey>,
+) -> Result<HashMap<String, Vec<u8>>, String> {
+    let query_resp = tool::json_client()
+        .call(&methods::query::RpcQueryRequest {
+            block_reference: BlockReference::Finality(Finality::Final),
+            request: QueryRequest::ViewState {
+                account_id: contract_id,
+                prefix: prefix.unwrap_or_else(|| vec![].into()),
+            },
+        })
+        .await
+        .map_err(|err| format!("Failed to query state: {:?}", err))?;
+
+    match query_resp.kind {
+        QueryResponseKind::ViewState(state) => Ok(tool::into_state_map(state.values)),
+        _ => Err("Could not retrieve access key".to_owned()),
+    }
+}
+
+pub async fn patch_state<T>(
+    account_id: AccountId,
+    key: String,
```

Comment:
This feels like a weird signature for patch state since if someone wants to patch multiple keys, they need to send a request for every single modification when the API supports patching in bulk. I really think this should take an `IntoIterator` where each item is a K, V. If you still want the convenience of a simpler API for one element, maybe another function could be considered which just uses this with [`iter::once`](https://doc.rust-lang.org/std/iter/fn.once.html) under the hood?

Reply:
In the js version we use a `RecordBuilder` to aggregate changes for a specific account.  But still have this API for a single update.

Reply:
@austinabell This ones actually tricky: taking an `IntoIterator` doesn't work for this case since we can't make a trait object out of `BorshSerialize` since it reference `self` in its methods. We can do `IntoIterator<Item=(Key, impl BorshSerialize)>` but now all patches have to be the same type in each iterable. I guess we'd end up needing something like what @willemneal suggested with the `RecordBuilder` potentially, but that requires exposing the developer to a lower level API, dealing directly with `StateRecord`s. I'm not sure what's the best way of handling this one

Reply:
Good point, probably use cases will require different value types. Perhaps the API is just `Item=(Key, Vec<u8>)` for the iterator and then functionality can be built on top of this, and you can keep the same signature for this patch single state function? We probably want to think of the best lower-level API, then think about what exactly to expose in this case.

Reply:
let's just keep the single state patch for now then. I'll make a separate ticket to address dealing the rest of this lower level API. Sounds good to you?